state/store.go

```diff
@@ -36,10 +36,47 @@ func calcABCIResponsesKey(height int64) []byte {
 	return []byte(fmt.Sprintf("abciResponsesKey:%v", height))
 }
 
+//----------------------
+
+type StoreI interface {
+	// LoadStateFromDBOrGenesisFile loads the most recent state.
+	// If the chain is new it will use the genesis file from the provided genesis file path as the current state.
+	LoadStateFromDBOrGenesisFile(string) (State, error)
```

Comment:
This isn't unique to this interface, but we should [avoid stuttering](https://golang.org/doc/effective_go.html#package-names). Since the package is named `state` it's already implied that operations are related to state, so e.g. `Load()` or `Save()` is probably sufficient. The same goes for e.g. `NewStateStore()` when `state.NewStore()` is already clear enough.

I'm fine with postponing this for a later cleanup though, and just migrating the methods as-is for now.